src/core/thread/dua_manager.cpp

```diff
@@ -342,9 +342,18 @@ void DuaManager::HandleBackboneRouterPrimaryUpdate(BackboneRouter::Leader::State
                                                    const BackboneRouter::Config &aConfig)
 {
     OT_UNUSED_VARIABLE(aConfig);
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    Mle::MleRouter &mle = Get<Mle::MleRouter>();
+#endif
 
     if (aState == BackboneRouter::Leader::kStateAdded || aState == BackboneRouter::Leader::kStateToTriggerRereg)
     {
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+        if (!mle.IsFullThreadDevice() && !mle.GetParent().IsThreadVersion1p1())
```

Comment:
Is the check `IsFullThreadDevice()` correct here? or do we need to check if role is child (`IsChild()`)?

e.g., an FTD may be attached as child to a parent that is 1.1.

Reply:
I see. Thanks. SGTM.